master/internal/api_workspace.go

```diff
@@ -281,6 +281,10 @@ func (a *apiServer) PostWorkspace(
 		return nil, status.Errorf(codes.InvalidArgument,
 			"name '%s' must be at most 80 character long", req.Name)
 	}
+	if len(strings.Trim(req.Name, " ")) == 0 {
```

Comment:
If we really want to restrict workspace names this way then we should be accounting for [all whitespace characters](https://en.wikipedia.org/wiki/Whitespace_character). There is a unicode utility to do this.

Reply:
probably something like this should trim both common whitespace and unicode separators:
```
var whitespaceText = regexp.MustCompile("[\\pZ[:space:]]")

func main() {
	s := "Hello, p\tl\naygr ou\u2028n\u00a0d"
	fmt.Println(whitespaceText.ReplaceAllLiteralString(s, ""))
}
```

Reply:
i think we just wanna trim left and right whitespaces to accept names like `Hello World` or `test v1`, so i made it like this

```go
len(strings.TrimFunc(name, unicode.IsSpace)) == 0
```

Reply:
okay, that should work. I did not realize we want to trim these spaces (what if user really wants to have them there). if we do, perhaps we'd want to do the same every time we parse workspace names in other APIs, e.g. in the experiment config on experiment submission.

Reply:
i see. then beckend just checks if the name isnt whitespace-only, and dont trim whitespace if there are at least 1 non-whitespace.